### PR TITLE
first pass at plan

### PR DESCRIPTION
**Pull request by willscott**
runs and validates the DKG process as a first test

### CODE REVIEW

File: ceremony/main.go [hunk 1]
```diff
@@ -42,28 +48,112 @@ func run(runenv *runtime.RunEnv) error {
...
+	config := &network.Config{
+		Network: "default",
+		Enable:  true,
+		Default: network.LinkShape{
+			Latency:   10 * time.Millisecond,
+			Bandwidth: 1 << 24,
+		},
+	}
 	config.IPv4 = &runenv.TestSubnet.IPNet
 	config.IPv4.IP = append(config.IPv4.IP[0:3], byte(seq))
```
Comment (raulk) [inline ceremony/main.go]: `seq` starts on 1, which means that you're probably requesting the gateway address here -- that's why the sidecar fails to allocate it.

**Review by nikkolasg — APPROVED** [review]
It's kinda hard for me to parse anything here - i dont have the bandwidth to dive in the testground documentation so I approve on trust :)

File: ceremony/main.go [hunk 2]
```diff
@@ -42,28 +48,92 @@ func run(runenv *runtime.RunEnv) error {
...
+		}
+	}
+	if !alive {
+		return fmt.Errorf("Node %d failed to start", seq)
+	}
+
+	var grp *key.Group
+	// run dkg
+	if isLeader {
+		client.Publish(ctx, leaderTopic, &LeaderAddr{leaderAddr})
```
Comment (yusefnapora) [inline ceremony/main.go:117]: Is this just used to synchronize the start times? If so I might replace with  `client.MustSignalAndWait(ctx, "dkg-start", runenv.TestInstanceCount)` before the `if` statement.

**Review by yusefnapora — APPROVED** [review]
Looks solid to me. Only thing that stood out is that we bail out if `runenv.TestSidecar` is false, so we should probably disable the `local:exec` runner in the manifest, since it won't have a sidecar.

File: ceremony/main.go (hunk 2 above)
Comment (willscott) [reply ceremony/main.go:117]: it's also that the leader needs to start before any of the others. by having that process signal and then startup, it seems to work better (though is technically still racy) than just having a single synchronization point

**Pull request merged**